test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1

```diff
@@ -0,0 +1,269 @@
+            if (!$isWindows) {
+                $result[0].Buffer.Count | Should -Be 0
+            } else {
+                $result[0].Buffer.Count | Should -Be 32
+            }
+        }
+}
+
+    # TODO: We skip the MTUSizeDetect tests on Unix because we expect 'TtlExpired' but get 'TimeOut' internally from .Net Core
+    Context "MTUSizeDetect" {
+        It "MTUSizeDetect works" -Pending:(!$isWindows) {
+            $result = Test-Connection $realName -MTUSizeDetect
+
+            $result | Should -BeOfType "System.Net.NetworkInformation.PingReply"
+            $result.Destination | Should -BeExactly $realName
+            $result.Status | Should -BeExactly "Success"
+            $result.MTUSize | Should -BeGreaterThan 0
+        }
+
+        It "Quiet works" -Pending:(!$isWindows) {
+            $result = Test-Connection $realName -MTUSizeDetect -Quiet
+
+            $result | Should -BeOfType "Int32"
+            $result | Should -BeGreaterThan 0
+        }
+    }
+
+    Context "TraceRoute" {
+        It "TraceRoute works" {
+            $result = Test-Connection $realName -TraceRoute
+            $replies = $result.Replies
+            # Check target host reply.
+            $pingReplies = $replies[-1].PingReplies
+
+            $result.Count              | Should -Be 1
+            $result                    | Should -BeOfType "Microsoft.PowerShell.Commands.TestConnectionCommand+TraceRouteResult"
+            $result.Source             | Should -BeExactly $hostName
+            $result.DestinationAddress | Should -BeExactly $realAddress
+            $result.DestinationHost    | Should -BeExactly $realName
+
+            $replies.Count               | Should -BeGreaterThan 0
+            $replies[0]                  | Should -BeOfType "Microsoft.PowerShell.Commands.TestConnectionCommand+TraceRouteReply"
+            $replies[0].Hop              | Should -Be 1
+
+            $pingReplies.Count           | Should -Be 3
+            $pingReplies[0].Address      | Should -BeExactly $realAddress
+            $pingReplies[0].Status       | Should -BeExactly "Success"
+            if (!$isWindows) {
+                $pingReplies[0].Buffer.Count | Should -Be 0
+            } else {
+                $pingReplies[0].Buffer.Count | Should -Be 32
+            }
+        }
+
+        It "Quiet works" {
+            $result = Test-Connection $realName -TraceRoute -Quiet
+
+            $result | Should -BeTrue
+        }
+    }
+    
+    Context "Connection" {
+        BeforeAll {
+            # Ensure the local host listen on port 80
+            $WebListener = Start-WebListener
+        }
+        
+        It "Test connection to local host port 80" {
+            Test-Connection '127.0.0.1' -TCPPort $WebListener.HttpPort | Should -BeTrue
+        }
+
+        It "Test connection to unreachable host port 80" {
+            Test-Connection $UnreachableAddress -TCPPort 80 -TimeOut 1 | Should -BeFalse
+        }
```

Comment:
doesn't look like this is indented properly

Reply:
Fixed.